### PR TITLE
Setup state management & caching

### PR DESCRIPTION
**Pull request by pratham15**
Added caching and state management to all payout items
Divided payout index pages into two components, a representational head & list of items
Removed unnecessary imports

### CODE REVIEW

File: atoms/index.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+export const proposalsAtom = atom(
+  {} as Record<number, CacheEntry<Payout<ProposalType>>>
+);
+
+export const bountiesAtom = atom(
+	{} as Record<number, CacheEntry<Payout<BountyType>>>
+);
+
+export const referralsAtom = atom(
+	{} as Record<number,CacheEntry< Payout<ReferralType>>>
```
Comment (siddharthborderwala) [inline atoms/index.ts]:
```ts
{} as Record<number, CacheEntry<Payout<ReferralType>>>
```

File: hooks/payout-hooks.ts [hunk 2]
```diff
@@ -0,0 +1,424 @@
...
+  }, [miscellaneous, from, limit]);
+
+  return {
+    data,
+    loading,
+    error,
+  } as const;
+};
+
+export const useMiscellanea: usePayoutHook<MiscellaneousType> = ({
```
Comment (siddharthborderwala) [inline hooks/payout-hooks.ts]: Ooh did not know Miscellaneous' plural is this, nice!

Shouldn't this one be useMiscelaneous and the above one be useMiscellanea?

File: hooks/payout-hooks.ts [hunk 3]
```diff
@@ -0,0 +1,424 @@
...
+        captureException(error);
+        setError(error);
+      } finally {
+        setLoading(false);
+      }
+    };
+
+    loadMiscellaneous();
+  }, [setMiscellaneous, contract, id]);
+  const data = useMemo(() => {
```
Comment (siddharthborderwala) [inline hooks/payout-hooks.ts]: leave a line before this

File: hooks/payout-hooks.ts [hunk 4]
```diff
@@ -0,0 +1,424 @@
...
+    if (referral[id]) return referral[id].data;
+    return undefined;
+  }, [referral, id]);
+
+  return {
+    data,
+    loading,
+    error,
+  } as const;
+};
```
Comment (siddharthborderwala) [inline hooks/payout-hooks.ts]: So this file has a lot of code repetition, there is a simple way to refactor all of this. Think about it and if you can't figure it out, ask me.

File: pages/dashboard/bounties/index.tsx [hunk 5]
```diff
@@ -45,39 +59,16 @@ const BountiesList: NextPage<WithContractChildProps> = ({ contract }) => {
...
-              description={p.description}
-              status={p.status}
-              proposer={p.proposer}
-              id={p.id}
-              link={`/dashboard/${Tabs.BOUNTIES}/${p.id}`}
-            />
-          ))
-        )}
-      </Box>
-      {bounties?.length == limit && (
```
Comment (siddharthborderwala) [inline pages/dashboard/bounties/index.tsx:68]: We need to put this block of code back in the component for pagination. Do this for the rest of the pages with list. I am sorry I forgot to mention this earlier

**Review by siddharthborderwala — CHANGES_REQUESTED** [review]
Good work from your side, not many changes required, except putting the pagination blocks back in.

You are getting better 🥳

File: hooks/payout-hooks.ts (hunk 2 above)
Comment (pratham15) [reply hooks/payout-hooks.ts]: Ah yes, I've reversed the names

**Review by siddharthborderwala — COMMENTED** [review]
Ok

**Review by siddharthborderwala — APPROVED** [review]

**Pull request merged**